docs: Minor improvements to GKE guide

pchaigno opened the pull request:
I found a couple nits with @qmonnet while using the GKE doc.

For the `global.restartPods` change (third commit), I'm not sure I got the right fix. The current version feels really frustrating though: we explain to the user that there's one more option we could have given helm at step n, and because we did not, we need to perform an additional step n+1. I'm guessing (?) the intent was to highlight that the pod restarts are optional and I tried to change the text accordingly.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10150)
<!-- Reviewable:end -->

pchaigno committed:
docs: Fix formatting of link to GCloud SDK

Link format was Markdown instead of reStructuredText.

Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
docs: Duplicate validation step

Since we deploy Cilium in its own cilium namespace, we need to change
the Validation section accordingly.

Signed-off-by: Paul Chaignon <paul@cilium.io>

rolinh commented on Documentation/gettingstarted/k8s-install-gke.rst:
```diff
@@ -70,23 +70,23 @@ Deploy Cilium release via Helm:
       --set nodeinit.removeCbrBridge=true \\
-      --set global.cni.binPath=/home/kubernetes/bin
+      --set global.cni.binPath=/home/kubernetes/bin \\
+      --set global.restartPods
```
This is intentionally omitted by default as restarting pods is a bit "aggressive". Maybe moving the note below before the `helm install ...` instruction rather than after would be a solution?

pchaigno replied:
Ah, okay, I understand. We're still asking users to restart all pods in the next section as part of the quick start guide though. Should we rework that next section? Is there a way to avoid the restarts?

pchaigno replied:
I've reworded a bit to try and clarify why `global.restartPods` is not set by default and moved that note before the actual command.

aanm commented on Documentation/gettingstarted/k8s-install-gke.rst:
```diff
@@ -70,23 +70,23 @@ Deploy Cilium release via Helm:
     kubectl create namespace cilium
     helm install cilium |CHART_RELEASE| \\
-      --namespace cilium \\
+      --namespace kube-system \\
```
Since we have removed problematic label that was causing cilium DS to be removed from the cluster, it is fine to have it in kube-system, what do you think @tgraf?

tgraf replied:
The addon manager will remove any DaemonSet which is not whitelisted.

tgraf replied:
We have duplicated the validate step for Azure CNI. We'll have to do the same for GKE for now.

pchaigno replied:
Ok. I'll do that.

pchaigno committed:
docs: Clarify why global.restartPods is not the default

Users might not want to restart pods as soon as the Cilium daemonset is
installed, so we provide a separate step for that. Since we also mention
the global.restartPods flag, let's clarify why that's not always a good
idea and not our default.

Signed-off-by: Paul Chaignon <paul@cilium.io>

rolinh commented on Documentation/gettingstarted/k8s-install-gke.rst:
```diff
@@ -66,6 +66,10 @@ Prepare & Deploy Cilium
 
+If you're ready to restart existing pods when initializing the node, you can
+also pass the ``--set global.restartPods`` flag to the ``helm`` command
+above. This will ensure all pods are managed by Cilium.
```
`s/above/below/`

pchaigno replied:
Arf! Fixed.

qmonnet commented on Documentation/gettingstarted/k8s-install-gke-validate.rst:
```diff
@@ -0,0 +1,50 @@
+    coredns-86c58d9df4-4g7dd                1/1     Running   0          13m
+    coredns-86c58d9df4-4l6b2                1/1     Running   0          13m
+
+Deploy the connectivity test
```
The connectivity test is the same for the three versions `Documentation/gettingstarted/k8s-install*validate.rst` (generic, Azure, GKE). Is it worth to extract it and to include it in each of those three files?

Whether we split the file or not, PR looks good to me.

pchaigno replied:
Agreed. I'll extract.

pchaigno committed:
docs: De-duplicate Connectivity Test section

Signed-off-by: Paul Chaignon <paul@cilium.io>
